File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.py
class Solution:
    def construct2DArray(self, original: List[int], m: int, n: int) -> List[List[int]]:
        leno, r, c = len(original), -1, 0
        if leno != m*n: return []

        matrix = [[0]*n for i in range(m)]
        for i in range(leno):
            if i % n == 0:
                r += 1
                c = 0
            matrix[r][c] = original[i]
            c += 1

        return matrix

File: 2022-convert-1d-array-into-2d-array/2022-convert-1d-array-into-2d-array.java
class Solution {
    public int[][] construct2DArray(int[] original, int m, int n) {
        int len = original.length, r = -1, c = 0;
        if(len != m*n) return new int[][]{};

        var matrix = new int[m][n];
        for(var i=0; i<len; i++){
            if(i % n == 0){
                r++;
                c = 0;
            }
            matrix[r][c++] = original[i];
        }

        return matrix;
    }
}